src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,383 @@
     if options.command == 'generate-manifest' or options.command == 'generate':
-        generate_file(os.path.join(cwd, 'gatk-preprocessing-manifest.tsv'), generate_manifest)
-
+        generate_file(os.path.join(cwd, 'manifest-toil-germline.tsv'), generate_manifest)
     # Pipeline execution
     elif options.command == 'run':
+        # Program checks
+        for program in ['curl', 'docker']:
+            require(next(which(program)), program + ' must be installed on every node.'.format(program))
+
         require(os.path.exists(options.config), '{} not found. Please run '
-                                             '"generate-config"'.format(options.config))
-        if not options.sample:
-            require(os.path.exists(options.manifest), '{} not found and no sample provided. Please '
-                                                       'run "generate-manifest"'.format(options.manifest))
-        # Parse config
-        parsed_config = {x.replace('-', '_'): y for x, y in yaml.load(open(options.config).read()).iteritems()}
-        inputs = argparse.Namespace(**parsed_config)
+                                                '"generate-config"'.format(options.config))
+        samples = []
         if options.manifest:
-            inputs.manifest = options.manifest
-    
-    inputs.cpu_count = multiprocessing.cpu_count() # FIXME: should not be called from toil-leader, see #186
-    inputs.memory = '15'
-    
-    Job.Runner.startToil(Job.wrapJobFn(batch_start, inputs, options.sample, options.output_dir, options.suffix), options)
+            samples.extend(parse_manifest(options.manifest))
+        if options.sample:
+            uuid, url, rg_line = options.sample.split()
+            samples.append((uuid, url, rg_line))
+
+        # Parse config
+        config = {x.replace('-', '_'): y for x, y in yaml.load(open(options.config).read()).iteritems()}
+        config_fields = set(config)
+        required_fields = {'genome.fa', 'mills', 'dbsnp', 'phase', 'hapmap', 'omni', 'output_dir',
+                           'run_bwa', 'preprocess', 'run_vqsr'}
+        require(config_fields > required_fields,
+                'Missing config parameters:\n{}'.format(', '.join(required_fields - config_fields)))
+
+        if config['output_dir'] is None:
+            config['output_dir'] = options.output_dir \
+                if options.output_dir else os.path.join(os.getcwd(), 'output')
+
+        if config['suffix'] is None:
+            config['suffix'] = options.suffix if options.suffix else ''
+
+        if not config['suffix'].startswith('.'):
+            config['suffix'] = '.' + config['suffix']
+
+        if config['synapse_id'] and config['synapse_pwd']:
+            config['syn'] = synapseclient.login(config['synapse_id'], config['synapse_pwd'])
+        else:
+            config['syn'] = None
+
+        if not config['xmx'].endswith('G'):
+            raise ValueError('xmx parameter must be in gigabytes (i.e. ends with "G"')
```

Comment:
This isn't a Java limitation. Are we parsing this to use for a memory requirement somewhere? If so, we should use the string to bytes code from BPL.